### PR TITLE
SyntaxNodes: subsume DotSelfExpr and ImplicitMemberExpr by MemberAccessExpr.

### PR DESCRIPTION
**Pull request by nkcsgexi**
Per our discussion in another PR, having these three similar nodes at
the same time  may unnecessarily complicate client side.

### CODE REVIEW

File: utils/gyb_syntax_support/ExprNodes.py [hunk 1]
```diff
@@ -297,27 +288,20 @@
...
-         children=[
-             Child('Expression', kind='Expr', is_optional=True),
-             Child('Dot', kind='Token',
+             Child("Dot", kind='Token',
                    token_choices=[
                        'PeriodToken', 'PrefixPeriodToken'
                    ]),
-             Child('SelfKeyword', kind='SelfToken'),
+             ## Name could be 'self'
+             Child("Name", kind='Token'),
```
Comment (jrose-apple) [inline utils/gyb_syntax_support/ExprNodes.py]: How do you distinguish `foo.self` and `` foo.`self` ``?

```code
struct Foo {
  var `self`: Int { return 0 }
}

let f = Foo()
print(f.self)
print(f.`self`)
```

Comment (jrose-apple) [reply utils/gyb_syntax_support/ExprNodes.py]: (I mean, yes, hopefully no one does this, but the compiler and language support it.)

Comment (nkcsgexi) [reply utils/gyb_syntax_support/ExprNodes.py]: `self` currently parsed as one token so the structure still applies. It parsed to be something like: `<MemberAccessExpr><IdentifierExpr>f</IdentifierExpr>.`self`</MemberAccessExpr>`

**Comment by nkcsgexi** [comment]
@swift-ci please smoke test

File: utils/gyb_syntax_support/ExprNodes.py (hunk 1 above)
Comment (jrose-apple) [reply utils/gyb_syntax_support/ExprNodes.py]: I see, `` `self` `` and `self` are different tokens, even though they have the same identifier text.

**Review by akyrtzi — APPROVED** [review]

**Comment by nkcsgexi** [comment]
@swift-ci please smoke test

**Comment by nkcsgexi** [comment]
@swift-ci please smoke test

**Pull request merged**